docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+# DCOS Python Style Guide
+###### Version 0.1
+###### April-6-2017
+
+For Python source code within DCOS, we prefer to keep Google Python Style Guide Revision 2.59 as the base and **extend** it with
+some changes that will make DCOS codebase look modern, consistent and easy to follow.
+
+##### Rationale
+
+With any style guide, the primary requirement is to adopt one and follow it consistently. With our choices below, we can use tools available in the ecosystem to our advantage, enforce consistency in the codebase, creating and publishing automatically generated documentation for our code.
+
+##### Versioned document
+
+Styleguide _conventions_ are not set in stone. We can improve upon this or remove inconvenient styles by creating a PR, discussing the merits or demerits to arrive at a consensus.
+
+#### Base Style Guide
+
+* [Google Python Style Guide Revision 2.59](https://google.github.io/styleguide/pyguide.html)
+
+#### Summary of Style Requirements DCOS which extend Base Style Guide and are specific to DCOS.
+
+* Import Requirements
+    * Adopt the base style guide preferences.
+
+        ```
+        Use import x for importing packages and modules.
+        Use from x import y where x is the package prefix and y is the module name with no prefix.
+        Use from x import y as z if two modules named y are to be imported or if y is an inconveniently long name.
+        ```
+    * Use `__init__.py` only to represent python packages.
+        * Avoid importing of objects in `__init__.py` for readability.
+* Import formatting
+
+    * The types of imports should be separated by single line and order is should be.
+        * standard library imports.
+        * third-party imports.
+        * application specific imports.
+
+    * The `from` style import and `import` statements should be separated by single line too.
+    * Each group of these imports should be sorted.
+
+For Example
+
+```
+import json
+import uuid
+
+from subprocess import CalledProcessError
+from typing import Callable, List
+
+import retrying
+
+from retrying import retry, RetryError
+from teamcity.messages import TeamcityServiceMessages
+
+import test_util.aws
+import test_util.cluster
+
+from pkgpanda.util import logger, write_string
+```
+
+* [Packages](https://google.github.io/styleguide/pyguide.html?showone=Packages#Packages)
+    * Import each module using the full pathname location of the module.
+    * No Relative Imports.
+
+* Maximum line length is 120 characters.
```

Comment:
Hi @vespian , I think, we can definitely aim for an unambiguous consistency here and perhaps simpler than Django rule as we can keep the col width same for both docstrings and code.

The 79 /80 col was width seems like a rule for teletypes, who screens were much smaller than any of our 15-inch, 16-inch and 21-inch laptop screen sizes.

* http://rdingwall.com/2008/05/31/is-the-80-character-line-limit-still-relevant/
* https://softwareengineering.stackexchange.com/questions/148677/why-is-80-characters-the-standard-limit-for-code-width
* The projects that follow might have had their origins in 1990s or early 2000's perhaps. They, for good reasons, do not want to change that style requirement.

Our current setting already enforces it the code to 120-chars( https://github.com/dcos/dcos/blob/master/tox.ini#L2) The files codebase within the project that are not with 120 col width, might need a reformat to 120 width. 

I think the tools only assert that you don't exceed 120 col width. It will be the responsibility of the developer and code-reviewer to keep the code appear consistently wrapped across the entire project.

Given current setting of 120 max-limit in flake8,  I thought _it is pretty non-controversial_ to pin that requirement in the style guide. Going back to 80 or 100 will be more controversial and **less desirable** IMO than keeping the current width requirement.

I hope this answers the concern.